xbmc/addons/kodi-addon-dev-kit/include/kodi/gui/controls/Rendering.h

```diff
@@ -0,0 +1,215 @@
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with KODI; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "../../AddonBase.h"
+#include "../Window.h"
+
+namespace kodi
+{
+namespace gui
+{
+namespace controls
+{
+
+  //============================================================================
+  ///
+  /// \defgroup cpp_kodi_gui_controls_CRendering Control Rendering
+  /// \ingroup cpp_kodi_gui
+  /// @brief \cpp_class{ kodi::gui::controls::CRendering }
+  /// **Window control for rendering own parts**
+  ///
+  /// This rendering control is used  when own parts are needed.  You  have  the
+  /// control  over them  to render direct  OpenGL or  DirectX  content  to  the
+  /// screen set by the size of them.
+  ///
+  /// Alternative  can be  the virtual  functions from t his been ignored if the
+  /// callbacks are  defined  by the  \ref CRendering_SetIndependentCallbacks function  and
+  /// class is used as single and not as a parent class.
+  ///
+  /// It has the header \ref Rendering.h "#include <kodi/gui/controls/Rendering.h>"
+  /// be included to enjoy it.
+  ///
+  /// Here you find the needed skin part for a \ref Addon_Rendering_control "rendering control"
+  ///
+  /// @note The  call of  the control is only  possible  from  the corresponding
+  /// window as its class and identification number is required.
+  ///
+
+  //============================================================================
+  ///
+  /// \defgroup cpp_kodi_gui_controls_CRendering_Defs Definitions, structures and enumerators
+  /// \ingroup cpp_kodi_gui_controls_CRendering
+  /// @brief **Library definition values**
+  ///
+
+  class CRendering : public CAddonGUIControlBase
+  {
+  public:
+    //==========================================================================
+    ///
+    /// \ingroup cpp_kodi_gui_controls_CRendering
+    /// @brief Construct a new control
+    ///
+    /// @param[in] window               related window control class
+    /// @param[in] controlId            Used skin xml control id
+    ///
+    CRendering(CWindow* window, int controlId)
+      : CAddonGUIControlBase(window)
+    {
+      m_controlHandle = m_interface->kodi_gui->window->get_control_render_addon(m_interface->kodiBase, m_Window->GetControlHandle(), controlId);
+      if (m_controlHandle)
+        m_interface->kodi_gui->control_rendering->set_callbacks(m_interface->kodiBase, m_controlHandle, this,
+                                              OnCreateCB, OnRenderCB, OnStopCB, OnDirtyCB);
+      else
+        kodi::Log(ADDON_LOG_FATAL, "kodi::gui::controls::CRendering can't create control class from Kodi !!!");
```

Comment:
Use __FUNCTION__ here and at the other places where you hard coded the funtion name for logging?